src/httpCodeLensProvider.ts

```diff
@@ -39,9 +39,14 @@ export class HttpCodeLensProvider implements CodeLensProvider {
             }
 
             if (blockStart <= blockEnd) {
+                let requestVariable = null;
+                if (blockStart > 0) {
+                    requestVariable = Selector.getRequestVariableDefinitionName(lines[blockStart - 1]);
+                }
                 const range = new Range(blockStart, 0, blockEnd, 0);
+                const requestLines = new RequestLines(range, document.uri.toString(), requestVariable);
```

Comment:
This seems you only add `requestVariable` when user trigger via the code lens action, if not, like using shortcut or _F1 Command_, no variables will be available. Since my extension also allows user just needs to click the mouse in the expected request range, and then trigger request via shortcut and _F1 command_

Reply:
Will look into this

Reply:
Should work both with F1 Commands, Code Lens and selecting text now.